src/pages/EnvironmentsPage/environmentsPermissions.ts

```diff
@@ -1,28 +1,35 @@
-import { LeoResourcePermissionsProvider } from 'src/analysis/Environments/Environments.models';
-import { getCreatorForCompute } from 'src/analysis/utils/resource-utils';
-import { App } from 'src/libs/ajax/leonardo/models/app-models';
+import { DecoratedComputeResource, PermissionsAndStateProvider } from 'src/analysis/Environments/Environments.models';
+import { getCreatorForCompute, isResourceDeletable } from 'src/analysis/utils/resource-utils';
+import { cromwellAppToolLabels } from 'src/analysis/utils/tool-utils';
+import { App, isApp } from 'src/libs/ajax/leonardo/models/app-models';
 import { PersistentDisk } from 'src/libs/ajax/leonardo/models/disk-models';
-import { ListRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+import { ListRuntimeItem, Runtime } from 'src/libs/ajax/leonardo/models/runtime-models';
 import { getTerraUser } from 'src/libs/state';
 
-const makePermissionsProvider = (userEmailGetter: () => string): LeoResourcePermissionsProvider => {
-  const permissionsProvider: LeoResourcePermissionsProvider = {
+const makeStateAndPermissionsProvider = (userEmailGetter: () => string): PermissionsAndStateProvider => {
+  return <PermissionsAndStateProvider>{
     canDeleteDisk: (disk: PersistentDisk) => {
       const currentUserEmail = userEmailGetter();
-      const canDelete = disk.auditInfo.creator === currentUserEmail;
-      return canDelete;
+      return disk.auditInfo.creator === currentUserEmail;
     },
     canPauseResource: (resource: App | ListRuntimeItem) => {
       const currentUserEmail = userEmailGetter();
       const creator = getCreatorForCompute(resource);
       return currentUserEmail === creator;
     },
+    canDeleteApp: (resource: DecoratedComputeResource) => {
+      if (isApp(resource)) {
+        return !Object.keys(cromwellAppToolLabels).includes(resource.appType);
+      }
+    },
+    canDeleteResource: (resourceType, resource: App | PersistentDisk | Runtime) => {
+      return isResourceDeletable(resourceType, resource);
+    },
   };
-  return permissionsProvider;
 };
 
 const currentUserEmailGetter = (): string => {
   return getTerraUser().email!;
 };
 
-export const leoResourcePermissions = makePermissionsProvider(currentUserEmailGetter);
+export const stateAndPermissionsProvider = makeStateAndPermissionsProvider(currentUserEmailGetter);
```

Comment:
again, I think the renames here are not needed, and the old names were more accurate... there is no state, it's just a permissions logic black-box.